magefile.go

```diff
@@ -650,24 +647,29 @@ func packageAgent(requiredPackages []string, packagingFn func()) {
 		defer os.Unsetenv(agentDropPath)
 
 		if devtools.ExternalBuild == true {
-			// for external go for all dependencies
-			dependencies := []string{
-				"auditbeat", "filebeat", "heartbeat", "metricbeat", "osquerybeat", "packetbeat", // beat dependencies
-				"apm-server",
+			externalBinaries := []string{
+				"auditbeat", "filebeat", "heartbeat", "metricbeat", "osquerybeat", "packetbeat",
 				// "cloudbeat", // TODO: add once working
 				"elastic-agent-shipper",
+				"apm-server",
 				"endpoint-security",
 				"fleet-server",
 			}
+
 			ctx := context.Background()
-			for _, beat := range dependencies {
-				for _, reqPackage := range requiredPackages {
+			for _, binary := range externalBinaries {
+				for _, platform := range platforms {
+					reqPackage := platformPackages[platform]
 					targetPath := filepath.Join(archivePath, reqPackage)
 					os.MkdirAll(targetPath, 0755)
-					newVersion, packageName := getPackageName(beat, version, reqPackage)
-					err := fetchBinaryFromArtifactsApi(ctx, packageName, beat, newVersion, targetPath)
+					newVersion, packageName := getPackageName(binary, version, reqPackage)
+					err := fetchBinaryFromArtifactsApi(ctx, packageName, binary, newVersion, targetPath)
 					if err != nil {
-						panic(fmt.Sprintf("fetchBinaryFromArtifactsApi failed: %v", err))
+						if strings.Contains(err.Error(), "object not found") {
+							fmt.Printf("Downloading %s: unsupported on %s, skipping\n", binary, platform)
```

Comment:
It seems to me too fragile just rely on the fact that the object isn't there to be sure the platform is unsupported. 

I believe the best would be the agent only try to download the supported binaries, and fail the build if they cannot be downloaded.

Reply:
This is hitting the official Elastic artifacts API, so if it isn't there it is a fairly strong indicator that the binary isn't supported. My first attempt at this was actually to encode which binaries were supported where into this magefile, but that is less likely to stay up to date than just trying to download the officially published artifacts themselves.

This path is only hit when building locally, it isn't actually used in the final release process which manually downloads each dependency from the DRA staging buckets. There's low risk of this causing us problems in the future.

Reply:
good point, I'm happy with this solution :)